rfc6962-bis.xml

```diff
@@ -481,26 +474,119 @@ Section 8.2 of <xref target="RFC3447"/>) using a key of at least 2048 bits.
+            case x509_sct: SignedCertificateTimestampDataV2;
+            case precert_sct: SignedCertificateTimestampDataV2;
+            case tree_head: TreeHeadDataV2;
+            case signed_tree_head: SignedTreeHeadDataV2;
+            case consistency_proof: ConsistencyProofDataV2;
+            case inclusion_proof: InclusionProofDataV2;
+        } data;
+        ItemExtension item_extensions&lt;0..2^16-1&gt;;
+    } TransItemV2;
 
+    struct {
+        Version version;
+        select (version) {
+            case v1: TransItemV1;
+            case v2: TransItemV2;
+        }
+    } TransItem;</artwork>
+        </figure>
+        <t>
+          <spanx style="verb">version</spanx> is the earliest version of this protocol to which the encapsulated data structure conforms. This document is v2. Note that <xref target="RFC6962">v1</xref> did not define <spanx style="verb">TransItem</spanx>, but this document specifies a mechanism (see <xref target="TransItemV1"/>) for v2 implementations to encapsulate existing v1 objects in the <spanx style="verb">TransItem</spanx> structure. Note also that, since each <spanx style="verb">TransItem</spanx> object is individually versioned, future revisions to this protocol could conceivably update some encapsulated data structures without having to update all of them.
+        </t>
+        <t>
+          <spanx style="verb">type</spanx> is the type of the encapsulated data structure. (Note that <spanx style="verb">TransType</spanx> combines the v1 type enumerations <spanx style="verb">LogEntryType</spanx>, <spanx style="verb">SignatureType</spanx> and <spanx style="verb">MerkleLeafType</spanx>). Future revisions of this protocol may add new <spanx style="verb">TransType</spanx> values.
+        </t>
+        <t>
+          <spanx style="verb">data</spanx> is the encapsulated data structure. The various structures named with the <spanx style="verb">DataV2</spanx> suffix are defined in later sections of this document.
+        </t>
+        <t>
+          <spanx style="verb">item_extension_type</spanx> identifies a single extension from the IANA registry in <xref target="item_extension_types"/>.
+        </t>
+        <t>
+          The interpretation of the <spanx style="verb">item_extension_data</spanx> field is determined solely by the value of the <spanx style="verb">item_extension_type</spanx> field. Each document that registers a new <spanx style="verb">item_extension_type</spanx> must describe how to interpret the corresponding <spanx style="verb">item_extension_data</spanx>.
+        </t>
+        <t>
+          <spanx style="verb">item_extensions</spanx> is a vector of 0 or more item extensions. This vector MUST NOT include more than one extension with the same <spanx style="verb">item_extension_type</spanx>. The extensions in the vector MUST be ordered by the value of the <spanx style="verb">item_extension_type</spanx> field, smallest value first.
```

Comment:
Because that's what Steve Kent wanted when we defined the top-level SCT extension syntax a few months ago.  ISTM that it made sense to define STH extensions and TransItem extensions in the same manner.
BTW, see http://www.ietf.org/mail-archive/web/trans/current/msg01388.html for some discussion of why Steve wanted the extensions to be ordered.